### PR TITLE
ENH: support index=True/False keyword for io.sql.get_schema

### PR DESCRIPTION
**Pull request by fangchenli**
- [x] closes #9084
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm, can you add a whatsnew note in Other Enhancements for 1.1.

`get_schema` is meant to be user-facing?

File: pandas/io/sql.py [hunk 1]
```diff
@@ -1869,7 +1869,12 @@ def get_schema(frame, name, keys=None, con=None, dtype=None):
     dtype : dict of column name to SQL type, default None
         Optional specifying the datatype for columns. The SQL type should
         be a SQLAlchemy type, or a string for sqlite3 fallback connection.
+    index : boolean, default: False
+         Whether to include the index of the DataFrame in the sql schema
+         New in version 1.1.0.
```
Comment (TomAugspurger) [inline pandas/io/sql.py:1874]:
```suggestion

        .. versionadded:: 1.1.0
```

File: doc/source/whatsnew/v1.1.0.rst [hunk 2]
```diff
@@ -329,6 +329,7 @@ Other enhancements
 - :meth:`DataFrame.to_excel` can now also write OpenOffice spreadsheet (.ods) files (:issue:`27222`)
 - :meth:`~Series.explode` now accepts ``ignore_index`` to reset the index, similarly to :meth:`pd.concat` or :meth:`DataFrame.sort_values` (:issue:`34932`).
 - :meth:`read_csv` now accepts string values like "0", "0.0", "1", "1.0" as convertible to the nullable boolean dtype (:issue:`34859`)
+- :meth:`~pandas.io.sql.get_schema` now accepts `index` parameter to include index of the DataFrame in the schema (:issue:`9084`).
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v1.1.0.rst:332]: This isn't in our API documentation. Can you add it to `doc/source/reference/io.rst` under the SQL section?

File: doc/source/reference/io.rst [hunk 3]
```diff
@@ -127,6 +127,13 @@ SQL
    read_sql_query
    read_sql
 
+.. currentmodule:: pandas.io.sql
```
Comment (TomAugspurger) [inline doc/source/reference/io.rst:130]: I think just do `sql.get_schema` in the block above. This shouldn't need its own section.

File: doc/source/whatsnew/v1.1.0.rst [hunk 4]
```diff
@@ -329,7 +329,6 @@ Other enhancements
 - :meth:`DataFrame.to_excel` can now also write OpenOffice spreadsheet (.ods) files (:issue:`27222`)
 - :meth:`~Series.explode` now accepts ``ignore_index`` to reset the index, similarly to :meth:`pd.concat` or :meth:`DataFrame.sort_values` (:issue:`34932`).
 - :meth:`read_csv` now accepts string values like "0", "0.0", "1", "1.0" as convertible to the nullable boolean dtype (:issue:`34859`)
-- :meth:`~pandas.io.sql.get_schema` now accepts `index` parameter to include index of the DataFrame in the schema (:issue:`9084`).
```
Comment (TomAugspurger) [inline doc/source/whatsnew/v1.1.0.rst]: We still want the release note.

**Review by TomAugspurger — APPROVED** [review]

**Comment by TomAugspurger** [comment]
Sorry, a couple issues with the docstring: https://github.com/pandas-dev/pandas/pull/35063/checks?check_run_id=842449220#step:10:9. Do you have time to fix those here? `python scripts/validate_docstring.py pandas.io.sql.get_schema` should print all the issues.

**Comment by jorisvandenbossche** [comment]
> get_schema is meant to be user-facing?

That's not entirely clear, see https://github.com/pandas-dev/pandas/issues/9960

**Comment by TomAugspurger** [comment]
I think we should just consider it public and document it as is.

**Comment by dsaxton** [comment]
@VirosaLi From the above looks like this might be ready to go? Maybe merge master again?

File: doc/source/whatsnew/v1.2.0.rst [hunk 5]
```diff
@@ -328,6 +328,7 @@ I/O
 - Bug in :func:`LongTableBuilder.middle_separator` was duplicating LaTeX longtable entries in the List of Tables of a LaTeX document (:issue:`34360`)
 - Bug in :meth:`read_csv` with ``engine='python'`` truncating data if multiple items present in first row and first element started with BOM (:issue:`36343`)
 - Removed ``private_key`` and ``verbose`` from :func:`read_gbq` as they are no longer supported in ``pandas-gbq`` (:issue:`34654`, :issue:`30200`)
+- :meth:``~pandas.io.sql.get_schema`` now accepts ``index`` parameter to include index of the DataFrame in the schema (:issue:`9084`).
```
Comment (dsaxton) [inline doc/source/whatsnew/v1.2.0.rst:331]:
```suggestion
- :meth:`~pandas.io.sql.get_schema` now accepts ``index`` parameter to include index of the DataFrame in the schema (:issue:`9084`).
```

**Comment by alimcmaster1** [comment]
@fangchenli - still interested in working on this or shall we close?

**Comment by jreback** [comment]
closing as stale. if you want to continue, pls ping and can re-open.

**Pull request closed without merging**